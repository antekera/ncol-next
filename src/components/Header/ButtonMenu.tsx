type ButtonMenuProps = {
  handleMenu: () => void
  isHeaderPrimary: boolean
  isMenuActive?: boolean
}

const MENU_TEXT = 'Menú'
const ARIA_LABEL = 'menú de categorías y búsqueda'

const ButtonMenu = ({
  isHeaderPrimary,
  handleMenu,
  isMenuActive
}: ButtonMenuProps) => {
  return (
    <button
      aria-haspopup='true'
      aria-expanded={isMenuActive}
      aria-label={ARIA_LABEL}
      type='button'
      onClick={handleMenu}
      className={`menu ease focus:shadow-outline flex items-center border-none pl-2 font-sans text-sm text-slate-700 duration-200 ${
        isHeaderPrimary
          ? 'text-zinc-100 hover:text-secondary'
          : 'hover:text-primary'
      }`}
    >
      <span className='hidden pr-2 md:block'>{MENU_TEXT}</span>
      <span className='material-symbols-rounded p2 cursor-pointer !text-4xl'>
        menu
      </span>
    </button>
  )
}

export { ButtonMenu }
