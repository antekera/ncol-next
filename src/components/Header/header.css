@layer components {
  .header {
    @apply text-white transition-all duration-300 ease-in;
  }

  .header-primary {
    @apply bg-primary md:min-h-[60px];
  }

  .header-not-single {
    @apply border-b border-slate-200 dark:border-neutral-500;
  }

  .header-primary-border {
    @apply border-dark-blue/20 text-white;
  }

  .header-not-primary {
    @apply text-zinc-400;
  }

  .header-not-share {
    @apply relative flex min-h-[60px] md:min-h-[90px] dark:bg-neutral-800;
  }

  .header-datetime {
    @apply text-zinc-400 border-l-2 border-zinc-300 pl-3 font-sans text-xs sm:py-2 md:pl-6;
  }
}