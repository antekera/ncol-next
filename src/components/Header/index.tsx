import { MenuIcon } from '@heroicons/react/outline'
import cn from 'classnames'
import { format } from 'date-fns'
import { es } from 'date-fns/locale'

import { Container, ProgressBar } from '../'
import { PAGE_DESCRIPTION } from '../../lib/constants'
import { Logo, LogoType } from '../Logo'
import { MainMenu } from './menu/Main'

const today: Date = new Date()

export enum HeaderType {
  Main = 'main',
  Category = 'category',
  Single = 'single',
  Share = 'share',
  Primary = 'primary',
}

type HeaderProps = {
  title?: string
  type?: 'main' | 'category' | 'single' | 'share' | 'primary'
  compact?: boolean
  isLoading?: boolean
  className?: string
  isMobile?: boolean
}

const defaultProps = {
  isMobile: false,
  compact: false,
  type: HeaderType.Main,
  title: PAGE_DESCRIPTION,
}

const Header = ({
  title,
  className,
  type,
  compact,
  isLoading,
}: HeaderProps) => {
  const isHeaderPrimary = type === HeaderType.Primary
  const isHeaderHome = type === HeaderType.Main
  const isHeaderShare = type === HeaderType.Share
  const isHeaderSingle = type === HeaderType.Single
  const isHeaderCategory = type === HeaderType.Category

  const headerClasses = cn(
    'flex transition-all ease-in delay-150 duration-300 text-white',
    { 'bg-primary': isHeaderPrimary },
    { 'bg-slate-200': isHeaderShare },
    { 'min-h-[60px] md:min-h-[90px]': !compact },
    { 'min-h-[60px] shadow-sm': compact },
    { 'border-b border-slate-200': !isHeaderSingle },
    { 'text-white': isHeaderPrimary },
    { 'text-zinc-400': !isHeaderPrimary },
    className
  )

  const logoMobile = {
    type: isHeaderPrimary ? LogoType.logonameb : LogoType.logoname,
    width: 140,
    height: 28,
  }

  const logoDesktop = {
    type: isHeaderPrimary ? LogoType.logocomb : LogoType.logocom,
    width: 220,
    height: 32,
  }

  return (
    <>
      <a href='#content' className='sr-only focus:not-sr-only'>
        Ir al contenido
      </a>
      <header className={headerClasses}>
        <Container className='flex items-center'>
          <div className='col'>
            <span className='md:hidden'>
              <Logo {...logoMobile} />
            </span>
            <span className='hidden md:block'>
              <Logo {...logoDesktop} />
            </span>
            <span className='sr-only'>{title}</span>
          </div>
          {isHeaderHome && (
            <div className='hidden pl-8 col sm:block'>
              <span className='py-2 pl-6 text-xs border-l-2 border-zinc-400'>
                Venezuela,
                <time>
                  {format(today, " dd 'de' MMMM 'de' yyyy", { locale: es })}
                </time>
              </span>
            </div>
          )}
          {isHeaderSingle && !isLoading && (
            <div className='hidden ml-8 col sm:block'>
              <p className='pl-6 mt-2 border-l-2 text-md md:text-xl border-zinc-400'>
                <a className='hover:text-primary ease duration-300' href='#'>
                  Costa Oriental
                </a>
              </p>
            </div>
          )}
          <div className='ml-auto col'>
            {isHeaderShare && !isLoading && '{Share Options}'}
            {!isHeaderShare && !isLoading && (
              <button
                aria-haspopup='true'
                aria-expanded='false'
                aria-label='menú de categorías y búsqueda'
                type='button'
                className={`flex items-center text-sm menu ease duration-300 text-slate-700 focus:shadow-outline pl-2 ${
                  isHeaderPrimary ? 'hover:text-white' : 'hover:text-primary'
                }`}
              >
                <span className='hidden pr-2 md:block'>Menú</span>
                <MenuIcon className='cursor-pointer w-7 h-7 p2' />
              </button>
            )}
          </div>
        </Container>
      </header>
      {isHeaderHome && isHeaderCategory && !isLoading && <MainMenu />}
      {isHeaderSingle && !isLoading && <ProgressBar percentage={70} />}
    </>
  )
}

Header.defaultProps = defaultProps

export { Header }
export type { HeaderProps }
