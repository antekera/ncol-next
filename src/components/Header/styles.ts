import { cn } from '@lib/shared'
import { HeaderType } from '.'

type HeaderClassesProps = {
  headerType?: string
  className?: string
}

export const getHeaderClasses = ({
  headerType,
  className
}: HeaderClassesProps) => {
  const isHeaderPrimary = headerType === HeaderType.Primary
  const isHeaderSingle = headerType === HeaderType.Single
  const isHeaderShare = headerType === HeaderType.Share

  return cn(
    'text-white transition-all duration-300 ease-in',
    { 'bg-primary md:min-h-[60px]': isHeaderPrimary },
    { 'border-b border-slate-200 dark:border-neutral-500': !isHeaderSingle },
    { 'border-dark-blue/20 text-white': isHeaderPrimary },
    { 'text-zinc-400': !isHeaderPrimary },
    {
      'relative flex min-h-[60px] md:min-h-[90px] dark:bg-neutral-800':
        !isHeaderShare
    },
    className
  )
}
