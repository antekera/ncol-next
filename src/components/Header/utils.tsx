import { LogoType } from '../Logo'

export const logoMobileOptions = (isHeaderPrimary: boolean) => {
  return {
    type: isHeaderPrimary ? LogoType.logonameb : LogoType.logoname,
    width: 140,
    height: 28,
  }
}

export const logoDesktopOptions = (isHeaderPrimary: boolean) => {
  return {
    type: isHeaderPrimary ? LogoType.logocomb : LogoType.logocom,
    width: 220,
    height: 32,
  }
}
