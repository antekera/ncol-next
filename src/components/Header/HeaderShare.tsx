import { Container } from '@components/Container'
import { Logo } from '@components/Logo'
import { ProgressBar } from '@components/ProgressBar'
import { Share as ShareOptions } from '@components/Share'
import { logoDesktopOptions, logoMobileOptions } from './utils'

type HeaderShareProps = {
  title?: string
  scrolled?: boolean
  isHeaderPrimary?: boolean
}

const HeaderShare = ({
  scrolled = false,
  title,
  isHeaderPrimary = false
}: HeaderShareProps) => {
  const logoMobile = logoMobileOptions(isHeaderPrimary)
  const logoDesktop = logoDesktopOptions(isHeaderPrimary)

  return (
    <header
      className={`fixed top-0 left-0 z-40 min-h-[55px] w-full bg-zinc-100 pt-3 text-slate-500 shadow-sm transition-all duration-300 ease-in-out md:min-h-[60px] dark:bg-neutral-800 dark:text-neutral-300 ${
        scrolled ? 'translate-y-0' : '-translate-y-16'
      }`}
    >
      <Container className='flex items-center'>
        <div className='col'>
          <span className='md:hidden'>
            <Logo {...logoMobile} />
          </span>
          <span className='hidden md:block'>
            <Logo {...logoDesktop} />
          </span>
          {title && <span className='sr-only'>{title}</span>}
        </div>
        <div className='col ml-auto whitespace-nowrap'>
          <ShareOptions />
        </div>
      </Container>
      <ProgressBar />
    </header>
  )
}

export { HeaderShare }
