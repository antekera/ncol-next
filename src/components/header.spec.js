import { render } from '@testing-library/react'
import Header from '../components/header'

describe('Header component', () => {
  test('should match snapshot', () => {
    const { container } = render(<Header />)
    expect(container).toMatchSnapshot()
  })
})
