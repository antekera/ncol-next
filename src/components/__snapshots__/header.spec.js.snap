// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
<div>
  <h2
    class="bg-primary mt-8 mb-20 text-2xl font-bold"
  >
    <a
      class="hover:underline"
      href="/"
    >
      Blog
    </a>
    .
  </h2>
</div>
`;
